docs/src/adr/007adr-restructuring.md

```diff
@@ -0,0 +1,161 @@
+# ADR-007: Apalache Package Structure Guidelines
+
+| author       | revision |
+| ------------ | --------:|
+| Jure Kukovec |        1 |
+
+This ADR documents the design policies guiding the package and dependency structure of Apalache.
+When introducing new classes, use the guidelines defined below to determine which package to place them in.
+
+## 1. Level Structure
+
+We define Apalache architecture in terms of enumerated levels (L0, L1, L2, etc.). Each level may only hold dependencies belonging to a lower level.
+
+Levels are split into two categories:
```

Comment:
Because, to my mind, "categories" suggested that we could have two parallel sets of levels, or that the categories were otherwise independent from the levels, but in fact they are just grouping of levels:

```suggestion
Levels are split into two strata:
```